client.go

```diff
@@ -163,14 +164,15 @@ func (c *client) Scan(s *hrpc.Scan) ([]*hrpc.Result, error) {
 		if err != nil {
 			return nil, err
 		}
+
 		scanres = res.(*pb.ScanResponse)
 		results = append(results, scanres.Results...)
 
 		// TODO: The more_results field of the ScanResponse object was always
 		// true, so we should figure out if there's a better way to know when
 		// to move on to the next region than making an extra request and
 		// seeing if there were no results
-		for len(scanres.Results) != 0 {
+		for len(scanres.Results) != 0 && uint32(len(results)) < numberOfRows {
```

Comment:
I agree that it should be redesigned to be able to control when to stop scanning. Something like [Scanner.nextRows()](http://opentsdb.github.io/asynchbase/javadoc/index.html) as in asynchbase should do it. 

At the moment, we are going to be limited to 128 rows in a scan, which seems incorrect.
However, If we are to remove the `uint32(len(results)) < numberOfRows` check, and someone decided to retrieve the whole key space of a table, the gohbase will do that and might run out of memory, which is also incorrect.

I'm ok with merging your fix, and then just working on redesigning the scan Api. Lets see what @tsuna thinks.